pkg/trace/api/dogstatsd.go

```diff
@@ -25,6 +26,13 @@ func (r *HTTPReceiver) dogstatsdProxyHandler() http.Handler {
 		})
 	}
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		body, err := io.ReadAll(req.Body)
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+		payloads := bytes.Split(body, []byte("\n"))
```

Comment:
Has it been confirmed that escaped new lines won't appear in normal statsd payloads? Otherwise this would split on `\\n` incorrectly right?

Reply:
Actually it looks like [event text](https://docs.datadoghq.com/developers/dogstatsd/datagram_shell/?tab=events) can have `\\n` to split the text, so this could be a problem. Will loop back around on this 👍

Reply:
https://go.dev/play/p/SblXAI_m5Ce just for posterity

Reply:
Also altered the test a bit just to make sure we're verifying this behavior :)

Reply:
Ah okay that makes sense. I guess I wasn't quite thinking of it right since `\\` is a character and `\n` is a character so we're all set :)

Reply:
It's confusing so don't worry. It took me a bit to wrap my head around it too 🤔